packages/plugin-ext/src/plugin/notification.ts

```diff
@@ -31,15 +32,18 @@ export class NotificationExtImpl implements NotificationExt {
         options: ProgressOptions,
         task: (progress: Progress<{ message?: string; increment?: number }>, token: CancellationToken) => PromiseLike<R>
     ): Promise<R> {
+        const id = new Deferred<string>();
+        const tokenSource = new CancellationTokenSource();
+        const progress = task({ report: async item => this.proxy.$updateProgress(await id.promise, item)}, tokenSource.token);
```

Comment:
Does it guarantee that the last item reported?

Reply:
@akosyakov 'you mean making the report function async could lead to out-of-order `$updateProgress` calls?

Reply:
it's the same promise they would be waiting for, shouldn't the event loop process them in order?

Reply:
yes, i think it should